extensions/amp-ooyala-player/0.1/amp-ooyala-player.js

```diff
@@ -103,31 +120,68 @@ class AmpOoyalaPlayer extends AMP.BaseElement {
 
   /** @override */
   pauseCallback() {
-    if (this.iframe_) {
+    // Only send pauseVideo command if the player is playing. Otherwise
+    // The player breaks if the user haven't played the video yet specially
+    // on mobile.
+    if (this.iframe_ && this.iframe_.contentWindow &&
+        this.playerState_ == PlayerStates.PLAYING) {
       this.pause();
     }
   }
 
+  /** @private */
+  handleOoyalaMessages_(event) {
+    if (event.origin != 'https://player.ooyala.com' ||
+        event.source != this.iframe_.contentWindow) {
+      return;
+    }
+    if (!event.data ||
```

Comment:
@mityaha Yes, <amp-ooyala> needs to dispatch `VideoEvents.PAUSE` and `VideoEvents.PLAYING` events when the underlying Ooyala player pauses or plays. This means you need to:
1- Have the iframed Ooyala player send a message back to the amp-ooyala component by using `postMessage` indicating that the video was paused/played.
2- amp-ooyala needs to listen to those messages and dispatch `VideoEvents.PAUSE` and `VideoEvents.PLAYING` events when receiving those messages. The YouTube code I referred you to does this but that code is very YouTube specific, so have to change it to work with your Ooyala specific messages.

Before running the tests, I recommend testing manually first to make sure the video-interface is implemented correctly. You can do this by adding `autoplay` to your example page and ensure
1- It autoplays on load.
2- It pauses when scrolling causes it to go out of view.
3- It plays again when scrolling brings it back to view.
4- The animated equalizer icon that's overlayed on your player by AMP is also paused/played when video is paused/played as result of scrolling. 5- ensure tapping/clicking on the autoplaing video "unmutes" it.